org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/JavaSemanticTokensProcessor.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Broadcom, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Broadcom, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.lsp4e.jdt;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jdt.ui.text.java.ISemanticTokensProvider;
+import org.eclipse.lsp4e.operations.semanticTokens.SemanticTokensDataStreamProcessor;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.SemanticTokensLegend;
+
+class JavaSemanticTokensProcessor {
+	
+	private final Function<Position, Integer> offsetMapper;
+	private final Function<String, ISemanticTokensProvider.TokenType> tokenTypeMapper;
+
+	/**
+	 * Creates a new instance of {@link SemanticTokensDataStreamProcessor}.
+	 *
+	 * @param tokenTypeMapper
+	 * @param offsetMapper
+	 */
+	@SuppressWarnings("restriction")
+	public JavaSemanticTokensProcessor(@NonNull final Function<String, ISemanticTokensProvider.TokenType> tokenTypeMapper,
+			@NonNull final Function<Position, Integer> offsetMapper) {
+		this.tokenTypeMapper = tokenTypeMapper;
+		this.offsetMapper = offsetMapper;
+	}
+
+	/**
+	 * Get the StyleRanges for the given data stream and tokens legend.
+	 *
+	 * @param dataStream
+	 * @param semanticTokensLegend
+	 * @return
+	 */
+	public @NonNull List<ISemanticTokensProvider.SemanticToken> getSemanticTokens(@NonNull final List<Integer> dataStream,
```

Comment:
the NonNull annotations are redundant since non-null is enabled by default for this package in package-info.java